*A small farm needs to make decision how much wheat and
how much barley they should grow.*
* The farm size is 40ha, the farm has 50,000€ cash reserves
* The production costs (ploughing, insecticides, fungicides,
fertiliser, harvesting) add up to 1,150€/ha for barley and
1,360€/ha for wheat.
* In 2018 the yield for wheat was 10t/ha, for barley 9t/ha.
* The expected price for wheat is 170€/t and for barley
170€/t.
* The price for wheat straw is 243€/ha and for barley straw is 300€/ha.

> What would be the optimal breakdown for wheat and
barley and the margin at the end of the year?

- Area of Wheat = $ X_w $
- Area of Barley = $ X_b $
- Farm size = 40 Hectors
- Reserves = EUR 50,000
- Barley Production Expenses ($ k_b $ )= 1150 EUR / ha
- Wheat Production Expenses ($ k_w $ ) = 1360 EUR / ha
- Income = 170 W + 170 B (per tonn)
- Income for straw = 243 W + 300 B (per hecter)

> We have to mazimise >> Hector for W + Hector for B <= 40 Hectors

> $ X_w +  X_b \le 40 $

*Depending upon the constraints of expenses and area, the equation could be written as :*

> $ k_w * X_w + k_b * X_b \le 50,000 $

In [44]:
import scipy.optimize as opt

In [45]:
cost_wheat = 1360 # EUR /  hactor
cost_barley = 1150 # EUR /  hactor

In [46]:
reserves = 50000
farm_size = 40

In [47]:
price_of_wheat = 170 # EUR /tonn
yield_of_wheat = 10 # tonn / hector
price_of_wheat_straw = 243 # EUR / hector
income_of_wheat = price_of_wheat * yield_of_wheat + price_of_wheat_straw # EUR / hector
profit_of_wheat =income_of_wheat - cost_wheat # EUR / hector
profit_of_wheat

583

In [48]:
price_of_barley = 150 # EUR /tonn
yield_of_barley = 9 # tonn/hector
price_of_barley_straw = 300 #EUR/hector

income_of_barley = price_of_barley * yield_of_barley + price_of_barley_straw
profit_of_barley = income_of_barley - cost_barley
profit_of_barley

500

**We need to maximise the following constraints:**

* Area to be utilized of the farm

> $ X_w + X_b \le 40 $

* reserves for the production cost

> $ 1360 X_w + 1150 X_b \le 50,000 $ 

**and we need to maximise the profit of wheat and barley**
> profit_of_wheat * Area in hector for wheat + profit_of_barley * Area in hector for barley

>> $ (profit  of  wheat) * X_w + (profit of barley) * X_b $



In [49]:
A = [[1,1] , [cost_wheat,cost_barley]]
b = [farm_size,reserves]
c = [-profit_of_wheat, -profit_of_barley]
bounds = [(0,farm_size),(0,farm_size)]

In [50]:
res = opt.linprog(c, A, b, bounds= bounds)
res

     con: array([], dtype=float64)
     fun: -21580.95238069383
 message: 'Optimization terminated successfully.'
     nit: 5
   slack: array([8.95894914e-10, 5.51081030e-07])
  status: 0
 success: True
       x: array([19.04761905, 20.95238095])

In [51]:
res.fun

-21580.95238069383

In [52]:
print(f"Using {res.x[0]:2.0f} hectors for wheat and \n {res.x[1]:2.0f} hectors for Barley, \n the profit is {- res.fun:2.0f}")

Using 19 hectors for wheat and 
 21 hectors for Barley, 
 the profit is 21581


| Reserves | Price of Wheat | Price of Barley | Area for Wheat | Area for Barley | Total Profit |
| --- | --- |--- | --- |--- | --- |
| 50,000 | 170 | 170 | 0 | 40 | 27200 |
| 50,000 | 170 | 150 | 19 | 21 | 21581 |